packages/dd-trace/src/appsec/remote_config/index.js

```diff
@@ -35,22 +35,33 @@ function enable (config) {
 function enableAsmData (appsecConfig) {
   if (rc && appsecConfig && appsecConfig.rules === undefined) {
     rc.updateCapabilities(RemoteConfigCapabilities.ASM_IP_BLOCKING, true)
-    rc.on('ASM_DATA', _asmDataListener)
+    rc.on('ASM_DATA', updateAsmData)
   }
 }
 
 function disableAsmData () {
   if (rc) {
-    rc.off('ASM_DATA', _asmDataListener)
+    rc.off('ASM_DATA', updateAsmData)
   }
 }
 
-function _asmDataListener (action, ruleData, ruleId) {
-  RuleManager.updateAsmData(action, ruleData, ruleId)
+function enableAsmDDRules (appsecConfig) {
+  if (rc && appsecConfig && appsecConfig.rules === undefined) {
+    rc.updateCapabilities(RemoteConfigCapabilities.ASM_DD_RULES, true)
+    rc.on('ASM_DD', updateAsmDDRules)
+  }
+}
+
+function disableAsmDDRules () {
+  if (rc) {
+    rc.off('ASM_DD', updateAsmDDRules)
```

Comment:
btw since we add the capability in `enableAsm...()` shouldn't we remove the capability when we disable it ? both for enableAsmData and enableAsmDDRules

Reply:
but are the disableAsm...  even called anywhere ? if not we can safely remove them and add them back once we need them don't you think ?

Reply:
yes! good catch